patches/api/0163-Make-the-default-permission-message-configurable.patch

```diff
@@ -43,20 +43,22 @@ index ca4e2d3b27f629e0d5e672fc915a5d03f0c0581d..17f8dd9870a47227a7c9bb09cceedb94
       * Creates a PlayerProfile for the specified uuid, with name as null
       * @param uuid UUID to create profile for
 diff --git a/src/main/java/org/bukkit/command/Command.java b/src/main/java/org/bukkit/command/Command.java
-index 7c80dc54776d0d66f7816b77136f6dbd9b801704..1994f15831de1ca1bb7b4f52c23567825766d3f9 100644
+index 7c80dc54776d0d66f7816b77136f6dbd9b801704..3f7f5017c8c1c0fc68cfc75c4c1ca87a60e7249f 100644
 --- a/src/main/java/org/bukkit/command/Command.java
 +++ b/src/main/java/org/bukkit/command/Command.java
-@@ -185,7 +185,12 @@ public abstract class Command {
+@@ -184,10 +184,11 @@ public abstract class Command {
+             return true;
          }
  
-         if (permissionMessage == null) {
+-        if (permissionMessage == null) {
 -            target.sendMessage(ChatColor.RED + "I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is a mistake.");
-+            // Paper start
-+            String bukkitPermissionMessage = Bukkit.getPermissionMessage();
-+            if (org.apache.commons.lang.StringUtils.isNotBlank(bukkitPermissionMessage)) {
-+                target.sendMessage(bukkitPermissionMessage);
-+            }
-+            // Paper end
-         } else if (permissionMessage.length() != 0) {
-             for (String line : permissionMessage.replace("<permission>", permission).split("\n")) {
+-        } else if (permissionMessage.length() != 0) {
+-            for (String line : permissionMessage.replace("<permission>", permission).split("\n")) {
++        // Paper start
++        String msg = this.permissionMessage != null ? this.permissionMessage : Bukkit.getPermissionMessage();
++        for (String line : msg.replace("<permission>", permission).split("\n")) {
++            if (org.apache.commons.lang.StringUtils.isNotBlank(line)) {
```

Comment:
For multiple lines, I feel like we should rather send it if does consist of a space: this can make for nicer messages such as:
```
<red>You are not authorised to perform this action!
 
<red>You need to purchase the rank <gold>Parrot Man</gold> to
<red>set your parrot companions.
```
with custom plugins to overwrite permission messages.

Alternatively, send every message that is not blank in its entirety.

Reply:
Oooh, very true ^_^